sync/src/synchronizer/block_fetcher.rs

```diff
@@ -131,14 +131,26 @@ impl BlockFetcher {
         let mut fetch = Vec::with_capacity(count);
         let mut start = fixed_last_common_header.number() + 1;
 
+        enum ShouldFetch {
+            Stored,
+            Inflightlimit,
+            Fetch,
+        }
+
         // Judge whether we should fetch the target block, neither stored nor in-flighted
         let mut should_fetch = |block_hash| {
             // NOTE: Filtering `BLOCK_STORED` but not `BLOCK_RECEIVED`, is for avoiding
             // stopping synchronization even when orphan_pool maintains dirty items by bugs.
             let stored = self
                 .active_chain
                 .contains_block_status(&block_hash, BlockStatus::BLOCK_STORED);
-            !stored && inflight.insert(self.peer, block_hash)
+            if stored {
+                ShouldFetch::Stored
+            } else if !inflight.insert(self.peer, block_hash) {
+                ShouldFetch::Inflightlimit
+            } else {
+                ShouldFetch::Fetch
+            }
```

Comment:
It can be simplified to

```
should_fetch -> (bool, bool) {
  return (stored, inflight)
}

match should_fetch() {
  (false, true) =>  // to fetch
  (false, false) => continue
  (true, _) => break;
}
```

Reply:
Too much bool causes difficulty in understanding, suggesting naming